internal/encryption/encryption.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+)
+
+type Encrypt struct {
+	gcm cipher.AEAD
+
+	logger logging.LoggerInterface
+}
+
+// Encrypt takes a plain string and returns a hex encoded string
+func (e *Encrypt) Encrypt(data string) (string, error) {
+	payload := []byte(data)
+	nonce, err := e.generateCipherNonce()
+	if err != nil {
+		e.logger.Errorf("error generating random the nonce, %v", err)
+		return "", fmt.Errorf("error generating random the nonce, %v", err)
+	}
+	ciphertext := e.gcm.Seal(nonce, nonce, payload, nil)
+	return hex.EncodeToString(ciphertext), nil
+}
+
+// Decrypt takes hex encoded string and returns the decrypted plain string
+func (e *Encrypt) Decrypt(hexData string) (string, error) {
+	encrypted, err := hex.DecodeString(hexData)
+	if err != nil {
+		e.logger.Errorf("error decoding hex encoded string, %v", err)
+		return "", fmt.Errorf("error decoding hex encoded string, %v", err)
+	}
+
+	noncePart, payloadPart := encrypted[:e.gcm.NonceSize()], encrypted[e.gcm.NonceSize():]
```

Comment:
can this end up in an out of range panic?
might be better to check the length of slice

Reply:
Nonce is fixed at the advised 12 units. But yes, it would be better to check for data length. I'll change this and will have to adjust / enhance tests also